designer/client/src/reducers/listReducer.tsx

```diff
@@ -132,6 +138,10 @@ export function listReducer(state: ListState, action: ListReducerActions) {
       selectedList.title = payload
       break
 
+    case ListActions.EDIT_TYPE:
+      selectedList.type = payload as ListTypeContent
```

Comment:
Would you mind if we moved the `as ListTypeContent` to the `onChange` handler?

We do something similar when changing page controller:

https://github.com/DEFRA/forms-designer/blob/98fe937bc3d076823c1707273f3830da2167d4ad/designer/client/src/PageCreate.tsx#L201-L207